heartbeat/scheduler/scheduler.go

```diff
@@ -105,27 +113,32 @@ func NewWithLocation(limit uint, location *time.Location) *Scheduler {
 }
 
 func (s *Scheduler) Start() error {
-	if s.running {
-		return errors.New("scheduler already running")
+	if !s.transitionRunning() {
+		return errors.New("scheduler already state")
```

Comment:
should the current state be part of this error?

Reply:
Yes, looks like there was a search and replace I did for `s/started/state` hence some weird grammar in a number of comments in this patch.